uniffi/src/scaffolding.rs

```diff
@@ -22,81 +22,62 @@ mod filters {
-            TypeReference::Optional(t) => format!("Option<{}>", ret_type_rs(t)?),
-            TypeReference::Sequence(t) => format!("Vec<{}>", ret_type_rs(t)?),
-            _ => panic!("[TODO: type_rs({:?})]", type_),
-        })
-    }
-
-    pub fn ret_type_rs(type_: &TypeReference) -> Result<String, askama::Error> {
-        Ok(match type_ {
-            TypeReference::U32 => "u32".to_string(),
-            TypeReference::U64 => "u64".to_string(),
-            TypeReference::Float => "f32".to_string(),
-            TypeReference::Double => "f64".to_string(),
-            TypeReference::Boolean => "bool".to_string(),
-            TypeReference::String => "String".to_string(),
-            TypeReference::Enum(name) => name.clone(),
-            TypeReference::Record(name) => name.clone(),
-            TypeReference::Optional(t) => format!("Option<{}>", ret_type_rs(t)?),
-            TypeReference::Sequence(t) => format!("Vec<{}>", ret_type_rs(t)?),
-            _ => panic!("[TODO: ret_type_rs({:?})]", type_),
+            Type::Int8 => "i8".to_string(),
+            Type::UInt8 => "u8".to_string(),
+            Type::Int16 => "i16".to_string(),
+            Type::UInt16 => "u16".to_string(),
+            Type::Int32 => "i32".to_string(),
+            Type::UInt32 => "u32".to_string(),
+            Type::Int64 => "i64".to_string(),
+            Type::UInt64 => "u64".to_string(),
+            Type::Float32 => "f32".to_string(),
+            Type::Float64 => "f64".to_string(),
+            Type::Boolean => "bool".to_string(),
+            Type::String => "String".to_string(),
+            Type::Enum(name) | Type::Record(name) | Type::Object(name) => name.clone(),
+            Type::Optional(t) => format!("Option<{}>", type_rs(t)?),
+            Type::Sequence(t) => format!("Vec<{}>", type_rs(t)?),
         })
     }
 
-    pub fn type_c(type_: &TypeReference) -> Result<String, askama::Error> {
+    pub fn type_ffi(type_: &FFIType) -> Result<String, askama::Error> {
         Ok(match type_ {
-            // Objects don't currently impl `ViaFfi`.
-            TypeReference::Object(_) => "u64".to_string(),
-            _ => format!("<{} as uniffi::support::ViaFfi>::Value", type_rs(type_)?,),
+            FFIType::Int8 => "i8".to_string(),
+            FFIType::UInt8 => "u8".to_string(),
+            FFIType::Int16 => "i16".to_string(),
+            FFIType::UInt16 => "u16".to_string(),
+            FFIType::Int32 => "i32".to_string(),
+            FFIType::UInt32 => "u32".to_string(),
+            FFIType::Int64 => "i64".to_string(),
+            FFIType::UInt64 => "u64".to_string(),
+            FFIType::Float32 => "f32".to_string(),
+            FFIType::Float64 => "f64".to_string(),
+            FFIType::RustBuffer => "ffi_support::ByteBuffer".into(),
+            FFIType::RustString => "*mut std::os::raw::c_char".into(),
+            FFIType::ForeignStringRef => "ffi_support::FfiStr<'_>".into(),
```

Comment:
Yes, good catch, this is left over from a previous factoring (and actually it's totally unused, because the only function that takes one of these is `string_free`, and that specifics its signature by hand.